torchtune/training/_distributed.py

```diff
@@ -57,26 +61,28 @@ class ParallelDims:
     dp_replicate: int
     dp_shard: int
     tp: int
+    cp: int
```

Comment:
More of a meta comment, but there has to be a better way to reason about all of these interactions than building a literal tree of if statements